pkg/nsx/services/subnetport/subnetport.go

```diff
@@ -0,0 +1,188 @@
+	if err != nil {
+		log.Error(err, "failed to create or update subnet port")
+		return err
+	}
+	err = service.SubnetPortStore.Operate(nsxSubnetPort)
+	if err != nil {
+		return err
+	}
+	log.Info("successfully updated NSX subnet port", "subnetPort", nsxSubnetPort)
+	return nil
+}
+
+func (service *SubnetPortService) GetPathFromSubnet(subnetName string, subnetNamespace string, client client.Client) (string, error) {
+	path := ""
+	subnetList := &v1alpha1.SubnetList{}
+	err := client.List(context.Background(), subnetList)
+	if err != nil {
+		log.Error(err, "failed to list all the subnet CRs")
+		return path, err
+	}
+	for _, subnet := range subnetList.Items {
+		if subnet.Name == subnetName {
+			path = subnet.Status.NSXResourcePath
+			if len(path) == 0 {
+				err = errors.New("empty NSX resource path from subnet")
+				return path, err
+			}
+			break
+		}
+	}
+	return path, nil
+}
+
+func (service *SubnetPortService) GetSubnetPortState(obj *v1alpha1.SubnetPort) (*model.SegmentPortState, error) {
+
+	// TODO: For now, only the parent subnet is supported.
+	specSubnet := obj.Spec.Subnet
+
+	// TODO: Get the subnet's NSX resource path from subnet store after the store is implemented.
+	nsxSubnetPath, err := service.GetPathFromSubnet(specSubnet, obj.Namespace, service.Client)
+	if err != nil {
+		log.Error(err, "failed to get NSX resource path from subnet", "subnet", specSubnet, "namespace", obj.Namespace)
+	}
+
+	paras := strings.Split(nsxSubnetPath, "/")
+	nsxOrgID := paras[2]
+	nsxProjectID := paras[4]
+	nsxVPCID := paras[6]
+	nsxSubnetID := paras[8]
+
+	nsxSubnetPortState, err := service.NSXClient.PortStateClient.Get(nsxOrgID, nsxProjectID, nsxVPCID, nsxSubnetID, string(obj.UID), nil, nil)
+	if err != nil {
+		// TODO: confirm the subnet port id format
+		log.Error(err, "failed to get subnet port state", "nsxSubnetPortID", obj.UID)
+		return nil, err
+	}
+	return &nsxSubnetPortState, nil
+}
+
+func (service *SubnetPortService) DeleteSubnetPort(obj *v1alpha1.SubnetPort) error {
+	nsxSubnetPort, err := service.buildSubnetPort(obj)
+	if err != nil {
+		log.Error(err, "failed to build Subnet")
+		return err
+	}
+	nsxSubnetPort.MarkedForDelete = &MarkedForDelete
+
+	// TODO: For now, only the parent subnet is supported.
+	specSubnet := obj.Spec.Subnet
+
+	// TODO: Get the subnet's NSX resource path from subnet store after the store is implemented.
+	nsxSubnetPath, err := service.GetPathFromSubnet(specSubnet, obj.Namespace, service.Client)
+	if err != nil {
+		log.Error(err, "failed to get NSX resource path from subnet", "subnet", specSubnet, "namespace", obj.Namespace)
+	}
+
+	paras := strings.Split(nsxSubnetPath, "/")
+	nsxOrgID := paras[2]
+	nsxProjectID := paras[4]
+	nsxVPCID := paras[6]
+	nsxSubnetID := paras[8]
+
+	err = service.NSXClient.PortClient.Delete(nsxOrgID, nsxProjectID, nsxVPCID, nsxSubnetID, *nsxSubnetPort.Id)
```

Comment:
for the delete operation, could we use the nsxPath get from SubnetPort CR status?

Reply:
done